src/hooks/useWithdraw.ts

```diff
@@ -0,0 +1,29 @@
+import { Asset, VTokenId } from 'types';
+import { useGetVTokenBalance, useRedeem, useRedeemUnderlying } from 'clients/api';
+
+export interface UseWithdrawParams {
+  amount: string;
+}
+
+const useWithdraw = ({ account, asset }: { account: string | undefined; asset: Asset }) => {
+  const { data: vTokenBalance } = useGetVTokenBalance(
+    { account, assetId: asset.id as VTokenId },
+    { enabled: !!account },
+  );
+  const { mutate: redeem, isLoading: isRedeemLoading } = useRedeem({
+    assetId: asset?.id as VTokenId,
+    account,
+  });
+  const { mutate: redeemUnderlying, isLoading: isRedeemUnderlyingLoading } = useRedeemUnderlying({
+    assetId: asset?.id as VTokenId,
+    account,
+  });
+  return {
+    redeem,
+    redeemUnderlying,
+    isLoading: isRedeemLoading || isRedeemUnderlyingLoading,
+    vTokenBalance,
+  };
```

Comment:
Given these mutations don't exchange results between one another, I don't think they should be merged into one hook, because doing that makes us lose functionalities (`onSuccess`, `onError` callbacks for example) as well as some of the results they return (errors for examples).